cmd/kubectl.go

```diff
@@ -37,34 +48,66 @@ func runKubectl(ctx *cli.Context) error {
 		return err
 	}
 
-	cluster, err := getClusterByID(c, c.UserConfig.FocusedCluster())
+	config, err := loadConfig(ctx)
 	if err != nil {
 		return err
 	}
 
-	config, err := c.ManagementClient.Cluster.ActionGenerateKubeconfig(cluster)
-	if err != nil {
-		return err
+	currentRancherServer := config.FocusedServer()
+	if currentRancherServer == nil {
+		return fmt.Errorf("no focused server")
 	}
 
-	tmpfile, err := ioutil.TempFile("", "rancher-")
+	currentToken := currentRancherServer.AccessKey
+	t, err := c.ManagementClient.Token.ByID(currentToken)
 	if err != nil {
 		return err
 	}
-	defer os.Remove(tmpfile.Name())
 
-	_, err = tmpfile.Write([]byte(config.Config))
+	currentUser := t.UserID
+	currentCluster := currentRancherServer.FocusedCluster()
+	filename := fmt.Sprintf("%s-%s-kubeconfig", currentUser, currentCluster)
+	home, err := os.UserHomeDir()
 	if err != nil {
 		return err
 	}
+	kubeconfigFilePath := fmt.Sprintf("%s/%s/%s", home, kubeconfigpath, filename)
```

Comment:
nit: Using `path.Join` or something similar might be better here.
https://golang.org/pkg/path/#Join

Reply:
Changed but is there any benefit of using that instead when the number of strings is already known apart from it being in that path package?

Reply:
I think it's a better practice because it takes care of weird edge cases (e.g. if `home`, `kubeconfigpath`, or `filename` have trailing slashes or something similar).